### PR TITLE
Kwarg Docs

### PR DESCRIPTION
**Pull request by Thetwam**
Added more detailed documentation for Keyword Arguments.

### CODE REVIEW

File: docs/keyword-args.rst [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+        search_term='John',
+        include=['email']
+    )
+
+Nested Parameters
+-----------------
+
+Some endpoints have parameters that look like :code:`foo[bar]`. Typically, there will be multiple parameters with the same prefix in the same endpoint.
+
+For example, :func:`canvasapi.account.Account.create_course` has several parameters that look like :code:`course[foo]`, as shown in the `Canvas Docs for Create a New Course <https://canvas.instructure.com/doc/api/courses.html#method.courses.create>`_. However, square brackets are not a valid character for python variables, so the following would **NOT** work:
```
Comment (jessemcbride) [inline docs/keyword-args.rst]: > However, square brackets are not a valid character for ~python~ Python variables

File: docs/keyword-args.rst [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+
+.. code:: python
+
+    # This is not valid, and will not work.
+    account.create_course(course[name]='Example Course')
+
+What Canvas is effectively doing with the bracket format is grouping things into objects. To get a similar effect in Python, CanvasAPI is set up to use dictionaries.
```
Comment (jessemcbride) [inline docs/keyword-args.rst]: > To get a similar effect in Python, CanvasAPI ~is set up to~ uses dictionaries.

File: docs/keyword-args.rst [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+
+Given an existing :class:`canvasapi.account.Account` object, :code:`account`, the parameter :code:`course[name]` can be passed like this:
+
+.. code:: python
+
+    account.create_course(course={'name': 'Example Course'})
+
+In the background, CanvasAPI will combine the keys of the dictionary with the keyword of the argument, and ultimately send the correct variable to Canvas.
+
+The benefit of this setup is that multiple parameters with the same prefix can be sent to the same keyword argument. So to pass the :code:`course[name]`, :code:`course[course_code]`, and :code:`course[is_public]` arguments, it would look like this:
```
Comment (jessemcbride) [inline docs/keyword-args.rst]: > The benefit of this ~setup~ pattern

File: docs/keyword-args.rst [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+            'is_public': True
+        },
+        enroll_me=True,
+        offer=False
+    )
+
+Complex Parameters
+------------------
+
+The three main types of parameters (Basic, List, and Nested) from above cover most types of parameters Canvas expects. However, there are some types that look deceptively more complex than they actually are. Below are some examples of how to handle these in CanvasAPI.
```
Comment (jessemcbride) [inline docs/keyword-args.rst]: > (basic, list, and nested)

File: docs/keyword-args.rst [hunk 5]
```diff
@@ -0,0 +1,149 @@
...
+
+Deep Nested Parameters
```
Comment (jessemcbride) [inline docs/keyword-args.rst]: > Deeply Nested

? Am okay with either

File: docs/keyword-args.rst [hunk 6]
```diff
@@ -0,0 +1,149 @@
...
+~~~~~~~~~~~~~~~~~~~~~~
+
+:func:`canvasapi.user.User.edit` has the parameter :code:`user[avatar][url]`, as shown in the `Canvas Docs for Edit a User <https://canvas.instructure.com/doc/api/users.html#method.users.update>`_. Any parameter that takes the form of :code:`foo[bar1][bar2]` with multiple bracketed sub-parameters follows the same rules as normal nested parameters, but additional level(s) deeper.
```
Comment (jessemcbride) [inline docs/keyword-args.rst]: > but  ~additional level(s) deeper~ with deeper nesting.

File: docs/keyword-args.rst [hunk 7]
```diff
@@ -0,0 +1,149 @@
...
+            'avatar': {
+                'url': 'http://example.com/john_avatar.png'
+            }
+        }
+    )
+
+List of Nested Parameters
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+:func:`canvasapi.account.Account.add_grading_standards` has the parameters :code:`grading_scheme_entry[][name]` and `grading_scheme_entry[][value]`, as shown in the `Canvas Docs for Create a New Grading Standard <https://canvas.instructure.com/doc/api/grading_standards.html#method.grading_standards_api.create>`_. Any parameter that takes the form of :code:`foo[][bar]` can be represented by a list of dictionaries.
```
Comment (jessemcbride) [inline docs/keyword-args.rst]: needs to be code:`grading_scheme_entry[][value]`.

Comment (Thetwam) [reply docs/keyword-args.rst]: Good catch. Fixed.

**Pull request merged**